### PR TITLE
Added pipenv config script for AuTest

### PR DESCRIPTION
**Pull request by rpm4118**
Pipenv provides a clean and simple way of creating a python virtual environment for AuTest. The new script will simply check for the existence of necessary components for pipenv without installing them, and if they do, run autest.

### CODE REVIEW

**Comment by dragon512** [comment]
[approve ci]

File: tests/Pipfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[[source]]
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+autest = "*"
```
Comment (dragon512) [inline tests/Pipfile]: Pin this to 1.7.3

File: tests/test-env-check.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    if [[ ${pytoutput:9:1} < 5 ]]
+    then
+        echo "Python 3 is too old. Need at least 3.5. "
+        return
+    else
+        echo "$pytoutput detected!"
+    fi
+fi
+
+# check for g++ -- for autest
```
Comment (zwoop) [inline tests/test-env-check.sh]: I'm not convinced we need this check on compilers, it's already done in configure.ac.

**Comment by dragon512** [comment]
@zwoop you want to keep the bootstrap.py around? I was hoping to remove that and have a simple pipfile that would install and manage the virtual env to run everything. 

I was going for only having a script ( or maybe have this in configure) that would check that the basic stuff we need is on the path ( ie python3.5 or better, g++ and pipenv) given that it use pipenv to easily  add, remove, setup the shell with what is needed

File: tests/test-env-check.sh (hunk 2 above)
Comment (dragon512) [reply tests/test-env-check.sh]: I believe he did this check as on some system the install os psutils needed for autest does not have a precompile .whl file for the system. not having g++ will mean autest will fail to install

File: tests/test-env-check.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# check for python3
+pytoutput=$(python3 --version 2>&1)
```
Comment (zwoop) [inline tests/test-env-check.sh]: Seems these rather convoluted checks could be replaced with something like:

```
python3 - << _END_
import sys

if sys.version_info.major < 3 or sys.version_info.minor < 5:
   exit(1)
_END_
```

and then check for $? != 1.

**Comment by zwoop** [comment]
Remove it. Just asking if we can retain the border and that comment input in, in the new configs.

**Comment by dragon512** [comment]
[approve ci]

**Comment by dragon512** [comment]
[approve ci]

**Comment by dragon512** [comment]
[approve ci]

**Comment by dragon512** [comment]
Need RAT fix.... @rpm4118 add the license header to the .sh files and pipfile you added

**Comment by dragon512** [comment]
[approve ci]

**Comment by zwoop** [comment]
Is this expected?

```
[root@6667b31860b3 /]# pipenv install
Creating a virtualenv for this project…
Pipfile: /Pipfile
Using /usr/bin/python3 (3.7.3) to create virtualenv…
⠹ Creating virtual environment...Already using interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /root/.local/share/virtualenvs/-x-v5uFv0/bin/python3
Also creating executable in /root/.local/share/virtualenvs/-x-v5uFv0/bin/python
Installing setuptools, pip, wheel...done.

✔ Successfully created virtual environment!
Virtualenv location: /root/.local/share/virtualenvs/-x-v5uFv0
Creating a Pipfile for this project…
Traceback (most recent call last):
  File "/usr/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv==2018.11.26', 'console_scripts', 'pipenv')()
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib64/python3.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/lib64/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib64/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.7/site-packages/pipenv/cli/command.py", line 254, in install
    editable_packages=state.installstate.editables,
  File "/usr/lib/python3.7/site-packages/pipenv/core.py", line 1741, in do_install
    pypi_mirror=pypi_mirror,
  File "/usr/lib/python3.7/site-packages/pipenv/core.py", line 609, in ensure_project
    validate=validate, skip_requirements=skip_requirements, system=system
  File "/usr/lib/python3.7/site-packages/pipenv/core.py", line 320, in ensure_pipfile
    project.create_pipfile(python=python)
  File "/usr/lib/python3.7/site-packages/pipenv/project.py", line 697, in create_pipfile
    config_parser = ConfigOptionParser(name=self.name)
  File "/usr/lib/python3.7/site-packages/pipenv/patched/notpip/_internal/cli/parser.py", line 142, in __init__
    assert self.name
AssertionError
```

**Comment by zwoop** [comment]
Assuming I'm right, we would need to persistently mount the /root/.local or something inside of Docker.

**Comment by dragon512** [comment]
I think you want to install autest when you create the image ( as you always have done), not when we run the build. The point for using docker here is that we can easily create environments with the exact setup we want and bring them up quickly for a build or to test some setup without having to install anything.

virtualenv should ignore everything installed in the system by default (it is like making a chroot) the only way around this is to use --system to install the packages in the system location or the --site-packages which I believe allows us to not ignore the system packages that are already installed. I have not tests these cases out directly as I was targeting work to make it easy for the devs to get a better working setup of autest locally.  more details here https://docs.pipenv.org/en/latest/advanced/

**Comment by zwoop** [comment]
Lets try it with

```
lrwxrwxrwx 1 jenkins jenkins 20 May 23 15:01 /home/jenkins/.local/share/virtualenvs -> /var/tmp/virtualenvs
```

where /var/tmp is persistent across docker instances (it's mounted -v /var/tmp:/var/tmp).

**Pull request merged**